tasks/Resources/RedHatMarketplace.resource

```diff
@@ -0,0 +1,46 @@
+*** Settings ***
+Documentation    Collections of keywords to register a cluster to RHM
+Library    OpenShiftLibrary
+Library    ../../venv/lib64/python3.8/site-packages/robot/libraries/OperatingSystem.py
+
+
+*** Variables ***
+${RH_MARKETPLACE_NS}=    openshift-redhat-marketplace
+
+
+*** Keywords ***
+Is Cluster Registered
+    [Documentation]    Returns TRUE if the cluster contains resources created by
+    ...                the registration process to RedHat Marketplace
+    ${result}=  Run Keyword And Return Status
+    ...  Run Keywords
+    ...  Oc Get  kind=Project  field_selector=metadata.name=${RH_MARKETPLACE_NS}  AND
+    ...  Oc Get  kind=Secret  namespace=${RH_MARKETPLACE_NS}
+    ...    field_selector=metadata.name=redhat-marketplace-pull-secret  AND
+    ...  Oc Get  kind=Subscription  namespace=${RH_MARKETPLACE_NS}
+    ...    field_selector=metadata.name=redhat-marketplace-operator
+    [Return]  ${result}
+
+Check Cluster Is Not Already Registered
+    [Documentation]    Checks the cluster is not registered. Fails if it is registered
+    [Arguments]    ${warn_on_failure}=${FALSE}
+    ${registered}=    Is Cluster Registered
+    IF    ${registered} == ${TRUE}
+        IF    ${warn_on_failure} == ${TRUE}
+            Log    message=Cluster may be already registered to RHM. Check the cluster...
+            ...    console=yes
+            ...    level=WARN
+        ELSE
+            Fail    msg=Cluster may be already registered to RHM. Check the cluster...
+        END
+    END
+
+Install RedHat Marketplace Operator
+    [Documentation]    Installs the RHM operator
+    Run    oc apply -f "https://marketplace.redhat.com/provisioning/v1/rhm-operator/rhm-operator-subscription?approvalStrategy=Automatic"
+
+Create RedHat Marketplace Secret
+    [Documentation]    Creates pull secret for fetching operators from RHM
+    [Arguments]    ${token}
+    Run    oc create secret generic redhat-marketplace-pull-secret -n openshift-redhat-marketplace --from-literal=PULL_SECRET=${token}
+    Run And Return Rc And Output    curl -sL https://marketplace.redhat.com/provisioning/v1/scripts/update-global-pull-secret | bash -s ${token}
```

Comment:
Is this output saved/used for anything?

Reply:
ah no, it's a mistake. Fixing it

Reply:
ah no, we need the output only for debug reasons, because the output of this curl will say something like "global pull secret patched" or similar. If I do only "Run" the logs won't show anything

Reply:
But then wouldn't it make sense to have a comparison/check on the output to confirm that it's what you expect to see?

Reply:
yes I can add it

Reply:
done

Reply:
oh I did and reverted the changes. Basically this kw is used in the Setup, and if it fails (cluster already registered for example), the installation wouldn't continue. That's why I didn't do it earlier. I would leave the output variable there only for debug